wolfcrypt/src/port/Renesas/renesas_tsip_util.c

```diff
@@ -3578,23 +3675,83 @@ WOLFSSL_LOCAL int tsip_SignRsaPkcs(wc_CryptoInfo* info, TsipUserCtx* tuc)
         else if (ssl->options.hashAlgo == sha256_mac)
             tsip_hash_type = R_TSIP_RSA_HASH_SHA256;
         else
-            ret = CRYPTOCB_UNAVAILABLE; 
+            ret = CRYPTOCB_UNAVAILABLE;
     }
-
+#else
+    (void)ssl;
+    
     if (ret == 0) {
+       if (tuc->sing_hash_type == md5_mac)
+           tsip_hash_type = R_TSIP_RSA_HASH_MD5;
+       else if (tuc->sing_hash_type == sha_mac)
+           tsip_hash_type = R_TSIP_RSA_HASH_SHA1;
+       else if (tuc->sing_hash_type == sha256_mac)
+           tsip_hash_type = R_TSIP_RSA_HASH_SHA256;
+       else
+           ret = CRYPTOCB_UNAVAILABLE;
+    }
+    
+    switch (tuc->wrappedKeyType) {
+        case TSIP_KEY_TYPE_RSA1024:
+            if (tuc->keyflgs_crypt.bits.rsapri1024_key_set != 1)
+            {
+                WOLFSSL_MSG("tsip rsa private key 2048 not set");
+                    ret = CRYPTOCB_UNAVAILABLE;
+            }
+            break;
+        case TSIP_KEY_TYPE_RSA2048:
+            if (tuc->keyflgs_crypt.bits.rsapri2048_key_set != 1)
+            {
+                WOLFSSL_MSG("tsip rsa private key 2048 not set");
+                    ret = CRYPTOCB_UNAVAILABLE;
+            }
+            break;
+        default:
+            WOLFSSL_MSG("wrapped private key is not supported");
+            ret = CRYPTOCB_UNAVAILABLE;
+            break;
+    }
+    
+#endif
+    
+    if (ret == 0) {
+       #ifdef WOLFSSL_RENESAS_TSIP_TLS
```

Comment:
Confirmed this macro guard change from "#ifndef WOLFSSL_RENESAS_TSIP_CRYPTONLY" to "#ifdef WOLFSSL_RENESAS_TSIP_TLS" solves the rsa certificate signature issue in TLS1.2.